web_src/js/features/imagediff.js

```diff
@@ -4,10 +4,12 @@ function getDefaultSvgBoundsIfUndefined(svgXml, src) {
   const DefaultSize = 300;
   const MaxSize = 99999;
 
-  const svg = svgXml.rootElement;
-
-  const width = svg.width.baseVal;
-  const height = svg.height.baseVal;
+  const svg = svgXml.documentElement;
+  const width = svg?.width?.baseVal;
+  const height = svg?.height?.baseVal;
+  if (width === undefined || height === undefined) {
+    return null; // in case some svg is invalid or doesn't have the width/height
```

Comment:
SVG without width/height is not invalid. The attributes default to `auto` which essentially means `100%` to fill the parent container:

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/width#svg

Not sure whether we should actually fill the container or default them to something more sane like `400px` maybe.

Reply:
What's your suggestion?

Reply:
Ideally I'd render with a max-width/max-height somewhere between 40% and 50% container width and render those size-less SVGs with that size as well. Maybe it can be done in CSS only, but JS-based solutions are also an option.

Reply:
Another simpler option may be to default to 300px width / 150px height as defined in the SVG spec here:

https://svgwg.org/specs/integration/#svg-css-sizing

Reply:
Could you help to propose a PR about the refactoring?

Reply:
I can check what can be done, but it would be helpful to have a proper testing repo. https://try.gitea.io/wxiaoguang/test/pulls/6/files seems unsuitable as is has a `namespace`-less SVG which will never properly render.

Reply:
Suiteable in the sense that it renders something, e.g. with namespace and some visible shape data.

Reply:
I can not understand why you need that .... the purpose of this PR is to handle invalid SVG gracefully.

If you work on a correct SVG, it's not related to this problem directly.

If you need some work for some special SVG, you can just prepare it ..........

Reply:
Yes, of course I can/will create test files myself.